modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableColumnHeader.java

```diff
@@ -701,6 +701,21 @@ protected void resizeColumnToFitContent(int maxRows) {
         }
     }
 
+    private <T> TableRow<T> createMeasureRow(TableView<T> tv, TableViewSkinBase tableSkin,
+            Callback<TableView<T>, TableRow<T>> rowFactory) {
+        TableRow<T> tableRow = rowFactory != null ? rowFactory.call(tv) : new TableRow<>();
+        tableSkin.getChildren().add(tableRow);
+        tableRow.applyCss();
+        if (!(tableRow.getSkin() instanceof SkinBase<?>)) {
+            tableSkin.getChildren().remove(tableRow);
+            // recreate with null rowFactory will result in a standard TableRow that will
+            // have a SkinBase-derived skin
+            tableRow = createMeasureRow(tv, tableSkin, null);
+        }
+        assert tableRow.getSkin() instanceof SkinBase<?>;
```

Comment:
Not sure if there is some convention about code asserts, maybe some else can tell. At least I never saw one in a PR here.

Reply:
I found assertions in various other parts of the code (e.g. `javafx.scene.control.skin.VirtualFlow<T>`) so I assumed this is ok.

Reply:
Some old code has assertions, but library code should not use them. Please delete them from any new / modified code.

Reply:
Deleted the assertion.